packages/vaadin-grid/src/vaadin-grid.d.ts

```diff
@@ -349,14 +349,30 @@ export interface GridEventMap<TItem> extends HTMLElementEventMap, GridElementEve
  */
 declare class GridElement<TItem = GridDefaultItem> extends HTMLElement {
   /**
+   * **Deprecated**
+   *
+   * See [`allRowsVisible`](#/elements/vaadin-grid#property-allRowsVisible)
+   *
    * If true, the grid's height is defined by its rows.
    *
    * Effectively, this disables the grid's virtual scrolling so that all the rows are rendered in the DOM at once.
    * If the grid has a large number of items, using the feature is discouraged to avoid performance issues.
+   *
    * @attr {boolean} height-by-rows
+   * @deprecated since version 14.7
    */
   heightByRows: boolean;
```

Comment:
Because of the confusing name, I've seen a lot of people "misusing" it because they thought that this means, the same as in V8 where you could do:

```java
grid.setHeightMode(HeightMode.ROW);
grid.setHeightByRows(3);
```

there should be a migration path somehow documented or mentioned. (I know it's hard because of the different height of rows)

Reply:
@knoobie As I understand it, this feature doesn't exist in the current web component, and as such that would be a new feature request? Just making sure because this PR is basically just about renaming the property so that there is less confusion to what it actually does.

Reply:
I'm totally in favor of the change! My comment is probably more about a note in the java components javadoc (but there was no PR atm.) just wanted to make sure it is mentioned somewhere.

And you are right! There is no feature in the web component for it.